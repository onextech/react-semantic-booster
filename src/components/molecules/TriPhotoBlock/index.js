import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Grid, Image } from 'semantic-ui-react';
import Block from '../../atoms/Block';
import { MediaCss } from '../../../utils/responsive';


const SecondImage = styled(Image)`
  ${MediaCss.min.sm`margin-top: 8em;`};
`;
const ThirdImage = styled(Image)`
  ${MediaCss.min.sm`margin-top: -6em;`};
`;

const TriPhotoBlock = ({ photos, block, grid }) => (
  <Block inverted vertical spacer={2} {...block}>
    <Grid container columns="equal" relaxed stackable {...grid}>
      <Grid.Row>
        <Grid.Column>
          <Image {...photos[0]} />
        </Grid.Column>
        <Grid.Column>
          <SecondImage {...photos[1]} />
        </Grid.Column>
      </Grid.Row>
      <Grid.Row>
        <Grid.Column mobile={16} tablet={8} computer={8}>
          <ThirdImage size="large" floated="right" {...photos[2]} />
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Block>
);

TriPhotoBlock.propTypes = {
  photos: PropTypes.arrayOf(PropTypes.object).isRequired, // semantic-ui image props
  block: PropTypes.object, // semantic-ui block props
  grid: PropTypes.object, // semantic-ui grid props
};

TriPhotoBlock.defaultProps = {
  block: undefined,
  grid: undefined,
};

export default TriPhotoBlock;
