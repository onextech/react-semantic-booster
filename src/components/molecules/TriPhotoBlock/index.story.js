import React from 'react';
import { storiesOf } from '@storybook/react';
import TriPhotoBlock from '.';
import { testImage1 } from '../../../../test/placeholders';

// eslint-disable-next-line import/prefer-default-export
export const photos = [
  {
    src: testImage1,
  },
  {
    src: testImage1,
  },
  {
    src: testImage1,
  },
];

storiesOf('TriPhotoBlock', module)
  .add('Default', () => (
    <TriPhotoBlock photos={photos} />
  ));
