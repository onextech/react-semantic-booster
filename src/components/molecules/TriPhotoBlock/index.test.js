import React from 'react';
import renderer from 'react-test-renderer';
import TriPhotoBlock from '.';
import { photos } from './index.story';

describe('<TriPhotoBlock>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<TriPhotoBlock photos={photos} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
