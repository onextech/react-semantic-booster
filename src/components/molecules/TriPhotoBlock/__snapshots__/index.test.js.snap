// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TriPhotoBlock> renders correctly 1`] = `
<div
  background={undefined}
  className="ui inverted vertical segment sc-bdVaJa jvNilU"
  height={undefined}
  spacer={2}
>
  <div
    className="ui container stackable relaxed equal width grid"
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <img
          className="ui image"
          src="https://upload.wikimedia.org/wikipedia/commons/c/cd/Panda_Cub_from_Wolong,_Sichuan,_China.JPG"
        />
      </div>
      <div
        className="column"
      >
        <img
          className="ui image sc-ifAKCX gTAWXr"
          src="https://upload.wikimedia.org/wikipedia/commons/c/cd/Panda_Cub_from_Wolong,_Sichuan,_China.JPG"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="eight wide computer sixteen wide mobile eight wide tablet column"
      >
        <img
          className="ui large right floated image sc-EHOje hzShKD"
          src="https://upload.wikimedia.org/wikipedia/commons/c/cd/Panda_Cub_from_Wolong,_Sichuan,_China.JPG"
        />
      </div>
    </div>
  </div>
</div>
`;
